torchbenchmark/score/compute_score.py

```diff
@@ -7,148 +7,151 @@
-
-    """
-    found_benchmarks = construct_benchmark_run_db(data)
-    weight_sum = 0.0
-    score_db = defaultdict(float)
-
-    for task, models in found_benchmarks.items():
-        for name, all_configs in models.items():
-            weight = b_weight[task] * (1.0/len(all_configs))
-            for mean, config, benchmark in all_configs:
-                weight_sum += weight
-                benchmark_score = weight * math.log(norm[benchmark] / mean)
-                score_db[config] += benchmark_score
-
-    assert abs(weight_sum - 1.0) < 1e-6, f"Bad configuration, weights don't sum to 1, but {weight_sum}"
-    return score_db
-
-def compute_score(weight_db, data, target, norm):
-    """
-    This API calculates the total score for all the benchmarks
-    that was run  by reading the data (.json) file.
-    The weights are then calibrated to the target score.
-
-    # TODO:
-    # score_db: score_per_config for now is just calculated.
-    # This needs to passed to plot_sweep.py to plot all the
-    # scores per config.
-    """
-    score = 0.0
-    score_db = get_score_per_config(data, weight_db, norm)
-
-    for config, scores in score_db.items():
-        score += scores
-        score_db[config] = target * math.exp(scores * 0.125)
-
-    score = target * math.exp(score)
-    return score
+class TorchBenchScore:
+    def __init__(self, spec=SPEC_FILE_DEFAULT, target=TARGET_SCORE_DEFAULT):
+        self.spec = spec
+        self.target = target
+        self.weights = None
+        self.norm = None
+
+    def setup_weights(self):
+        """
+        Calculates the static benchmark weights by iterating the spec
+        file and constructs a dictionary with (key, value) pair
+        is (task, weight_for_benchmark_per_task)
+
+        The spec here can be user defined .yaml file or if no spec is defined,
+        the default score.yaml is used.
+        """
+        # Load the spec file
+        with open(self.spec) as spec_file:
+            spec = yaml.full_load(spec_file)
+
+        self.weights = defaultdict(float)
+        category_spec = spec['hierarchy']['model']
+        domain_weight = 1.0/ len(category_spec)
+        for domain in category_spec:
+            tasks = category_spec[domain]
+            task_weight = 1.0 / len(tasks)
+            for task in tasks:
+                benchmarks = tasks[task]
+                benchmark_weight = 1.0 / len(benchmarks)
+                self.weights[task] = domain_weight * task_weight * benchmark_weight
+
+    def setup_benchmark_norms(self, data):
+        """
+        Helper function which gets the normalization values per benchmark
+        by going through the reference data file.
+
+        This reference data can be user provided or by default the first file
+        in the directory of data files is considered as the reference data file
+        """
+        self.norm = {b['name']: b['stats']['mean'] for b in data['benchmarks']}
+
+    def _get_model_task(self, model_name):
```

Comment:
this probably shouldn't be a function of the TorchBenchScore class, seems like a standalone func

Reply:
Ok, agreed. Moved from the class